marklogic-data-hub/src/test/java/com/marklogic/hub/entity/EntityServicesAlignmentTest.java

```diff
@@ -42,10 +47,85 @@ public void testDeployTDEWithNoEntities() {
+        String entity = "{\n" +
+            "  \"info\": {\n" +
+            "    \"title\": \"Customer\",\n" +
+            "    \"version\": \"0.0.1\",\n" +
+            "  },\n" +
+            "  \"definitions\": {\n" +
+            "    \"Customer\": {\n" +
+            "      \"properties\": {\n" +
+            "        \"customerId\": {\n" +
+            "          \"datatype\": \"integer\",\n" +
+            "        },\n" +
+            "        \"name\": {\n" +
+            "          \"datatype\": \"string\",\n" +
+            "          \"collation\": \"http://marklogic.com/collation/codepoint\"\n" +
+            "        }\n" +
+            "      }\n" +
+            "    }\n" +
+            "  }\n" +
+            "}";
+        File entityFile = getHubProject().getHubEntitiesDir().resolve("Customer.entity.json").toFile();
+        try {
+            mapper.writerWithDefaultPrettyPrinter().writeValue(entityFile, mapper.readTree(entity));
+            installUserArtifacts();
+            assertTrue(false, "Inserting entity without baseUri should fail");
+        } catch (IOException e) {
+            logger.info("Cannot insert entity without baseUri");
+        }
+        finally {
+            FileUtils.deleteQuietly(entityFile);
+        }
+        assertEquals(0, getDocCount(HubConfig.DEFAULT_FINAL_SCHEMAS_DB_NAME, TDE_COLLECTION));
+        assertEquals(0, getDocCount(HubConfig.DEFAULT_STAGING_SCHEMAS_DB_NAME, TDE_COLLECTION));
+    }
 
+    @Test
+    public void testDeployInvalidEntity() {
+        String entity = "{\n" +
+            "  \"info\": {\n" +
+            "    \"title\": \"Customer\",\n" +
+            "    \"version\": \"0.0.1\",\n" +
+            "  },\n" +
+            "  \"definitions\": {\n" +
+            "    \"Customer\": {\n" +
+            "      \"properties\": {\n" +
+            "        \"customerId\": {\n" +
+            "          \"datatype\": \"integer\",\n" +
+            "        },\n" +
+            "        \"name\": {\n" +
+            "          \"datatype\": \"string\",\n" +
+            "          \"collation\": \"http://marklogic.com/collation/codepoint\"\n" +
+            "        },\n" +
+            "        \"shipping\": {\n" +
+            "          \"datatype\": \"array\",\n" +
+            "          \"items\": {\n" +
+            "            \"$ref\": \"#/definitions/Address\"\n" +
+            "          }\n" +
+            "        }\n" +
+            "      }\n" +
+            "    }\n" +
+            "  }\n" +
+            "}";
+        File entityFile = getHubProject().getHubEntitiesDir().resolve("Customer.entity.json").toFile();
+        try {
+            mapper.writerWithDefaultPrettyPrinter().writeValue(entityFile,  mapper.readTree(entity));
+            installUserArtifacts();
+            assertTrue(false, "Inserting invalid entity should fail");
+        } catch (IOException e) {
+            logger.info("Cannot insert invalid entity");
```

Comment:
For catch expected exceptions, I recommend: "Caught expected exception due to invalid entity: " + e.getMessage(). 

In fact, it's preferable to make an assertion on the contents of the message so you got the expected error. I'm surprised an IOException is being thrown - don't we want to catch something like an ML FailedRequestException?